`derive_arbitrary` build failure with with `generate-lockfile -Z minimal-versions`

It seems like I may not have understood the actual problem in #134.

I think the actual issue is that `derive_arbitrary` is not specifying the minimal `syn` version it can build with.

I'm encountering the same problem (and a new one) on the latest release:

https://github.com/RustCrypto/formats/actions/runs/3832620440/jobs/6523151884

```
error[E0599]: no function or associated item named `into_compile_error` found for struct `Error` in the current scope
  --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/derive_arbitrary-1.2.2/src/lib.rs:16:37
   |
16 |         .unwrap_or_else(syn::Error::into_compile_error)
   |                                     ^^^^^^^^^^^^^^^^^^
   |                                     |
   |                                     function or associated item not found in `Error`
   |                                     help: there is a method with a similar name: `to_compile_error`

error[E05[99](https://github.com/RustCrypto/formats/actions/runs/3832620440/jobs/6523151884#step:9:100)]: no method named `len` found for reference `&Fields` in the current scope
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/derive_arbitrary-1.2.2/src/lib.rs:260:38
    |
260 |                 if idx + 1 == fields.len() {
    |                                      ^^^ method not found in `&Fields`
```

--- a/derive/Cargo.toml
+++ b/derive/Cargo.toml
@@ -21,7 +21,7 @@ rust-version = "1.63.0"
 [dependencies]
 proc-macro2 = "1.0"
 quote = "1.0"
-syn = { version = "1.0", features = ['derive'] }
+syn = { version = "1.0.56", features = ['derive'] }
 
 [lib]
 proc_macro = true

